unity/Assets/Scripts/BaseFPSAgentController.cs

```diff
@@ -2960,14 +2963,21 @@ int layerMask
                     //make sure to rotate just the Camera, not the whole agent
                     m_Camera.transform.LookAt(targetSOP.transform, transform.up);
 
-                    var visibleSimObjects = this.GetAllVisibleSimObjPhysics(this.maxVisibleDistance);
+                    bool isVisible = false;
+                    if (this.visibilityScheme == VisibilityScheme.Distance) {
+                        isVisible = isSimObjVisible(this.m_Camera, targetSOP, this.maxVisibleDistance, planes);
+                    } else {
+                        var visibleSimObjects = this.GetAllVisibleSimObjPhysics(this.maxVisibleDistance);
+                        isVisible = visibleSimObjects.Any(sop => sop.objectID == targetSOP.objectID);
```

Comment:
Have read around that it is not recommended to run Linq functions in frequent running functions in unity as it is quite slow, ok for initialization etc. but not great for update. 

Have not measured it so I am not sure, and regardless I vote for the new visibility scheme